drivers/usb/source/USBMIDI.cpp

```diff
@@ -126,7 +126,7 @@ bool USBMIDI::write(MIDIMessage m)
 
         _flags.clear(FLAG_WRITE_DONE);
         USBDevice::write_start(_bulk_in, buf, 4);
-        uint32_t flags = _flags.wait_any(FLAG_WRITE_DONE | FLAG_DISCONNECT, osWaitForever, false);
+        uint32_t flags = _flags.wait_any(FLAG_WRITE_DONE | FLAG_DISCONNECT | FLAG_CONNECT, osWaitForever, false);
```

Comment:
This doesn't quite make sense to me. How can you have started the write (with `USBDevice::write_start()` without being connected? Should we instead check the return value from `USBDevice::write_start()` for error before starting this wait? I could imagine the wait getting stuck if `write_start()` failed.

Reply:
Hi @Patater 

I have verified return of `USBDevice::write_start()` and it is `true`,  there is no error from this api. 

Looks like EventFlags not updating FLAG_WRITE_DONE flag.

The driver hangs here 

https://github.com/ARMmbed/mbed-os/blob/efaf159ca44f62ca665f50312a410360a85abc70/cmsis/CMSIS_5/CMSIS/RTOS2/RTX/Source/rtx_evflags.c#L565